File: server/enums/support_status.py
"""
Support ticket status enumeration.
"""
from enum import Enum


class SupportTicketStatus(str, Enum):
    """
    Enumeration of possible statuses for a support ticket.
    """

    OPEN = "open"
    WAITING_USER = "waiting_user"
    WAITING_SUPPORT = "waiting_support"
    RESOLVED = "resolved"
    CLOSED = "closed"


__all__ = ["SupportTicketStatus"]

File: web/app/constants/supportStatus.ts
import type { StatusPresentation } from '~/types/StatusPresentation'

export const SUPPORT_STATUS_PRESENTATION: Record<string, StatusPresentation> = {
  open: { label: 'Ouvert', badgeClass: 'app-badge--progress' },
  waiting_support: { label: 'Attente support', badgeClass: 'app-badge--progress' },
  waiting_user: { label: 'Attente client', badgeClass: '' },
  resolved: { label: 'Résolu', badgeClass: 'app-badge--success' },
  closed: { label: 'Fermé', badgeClass: '' },
}
